include/swift/AST/Type.h

```diff
@@ -592,6 +592,13 @@ class CanGenericSignature {
   GenericSignature *getPointer() const {
     return Signature;
   }
+
+  bool operator==(const swift::CanGenericSignature& other) {
+    return Signature == other.Signature;
+  }
+  bool operator==(const swift::CanGenericSignature* other) {
```

Comment:
This is generally not an overload we want to support; `==` comparisons should always be symmetric.

Reply:
Perhaps explicitly converting from the `CanGenericSignature` to the `GenericSignature` at the sites is the way to go then?

Reply:
Makes sense, I've removed that overload. In the process of replacing the call site, it turns out that in
```
  // Emit the raw application.
  auto genericSig =
    fn.getType().castTo<SILFunctionType>()->getGenericSignature();
```
MSVC deduces that genericSig is a `GenericSignature`, not a `GenericSignature*` as would be expected. Specifying the type seems to fix it...